public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx

```diff
@@ -224,12 +232,14 @@ export const ConnectionSVG = ({
 
               if (index === 0) {
                 // For first vertex
-                addVertices.push(calculateMidpoint(0, 0, x, y));
+                addVertices.push(
+                  calculateMidpoint((x1 - xStart) / (xEnd - xStart), (y1 - yStart) / (yEnd - yStart), x, y)
```

Comment:
might be a little cleaner to break out these operations and assign them to a variable to be passed into this helper function

both here and [line 331 of this file](https://github.com/grafana/grafana/pull/85443/files#diff-88e95aa7a0e12a09f7ec48f2ad2f5289e30d62f438e5948897c3b18e1d86ca1dR331)

Reply:
Good suggestion! I'll be sure to incorporate this during clean up work: https://github.com/grafana/grafana/issues/85013